### ISSUE TITLE
The executable X does not contain bitcode - tvOS

### ISSUE THREAD

**Issue by birarda** [Carthage/Carthage#915]
I don't think this is a Carthage issue, but I'm guessing I'll get the most helpful responses here given the recent discussion on how to handle bitcode support in frameworks. 

I'm trying to submit a build to iTunes Connect but am getting back an email that all of the frameworks I'm using via Carthage do not contain bitcode. I've confirmed that two of them should (via `BITCODE_GENERATION_MODE`) and have manually added `BITCODE_GENERATION_MODE` to the others (and then re-run carthage build).

I'm guessing I've missed something.

**Comment by NachoSoto** [comment]
Probably a silly question, but I imagine they also have `BITCODE_ENABLED` set to `true`?

**Comment by birarda** [comment]
Yep, checked them all.

**Comment by NachoSoto** [comment]
Another silly question :P when you submitted the archive, you checked the box that says "submit bitcode", correct?
This is probably even a sillier question, because I think Xcode doesn't actually give you the option for `tvOS`, but just making sure.

Are you using Xcode's UI to submit your app, or CLI?

**Comment by birarda** [comment]
Xcode's UI.

I don't think I was even given an option to enable bitcode - either on upload or in the build settings for the project.

**Comment by birarda** [comment]
Do I need BITCODE_GENERATION_MODE on my project as well, or just on the projects whose frameworks I am using?

**Comment by birarda** [comment]
Doing some more reading it seems that I do not need that, nor do the frameworks I am using - they just need "Enable Bitcode" set to YES in their project settings?

Interesting to note that the frameworks produced in my build folder do fail the test outlined here - http://stackoverflow.com/questions/32808642/how-to-check-if-a-framework-is-bitcode-supported-for-xcode7

**Comment by birarda** [comment]
FWIW I would like to use Carthage but wanted to get a build up today and tried with the same frameworks via Cocoapods and the upload was validated. Happy to provide whatever additional information I can.

**Comment by ikesyo** [comment]
Could you share your Cartfile entries and `carthage version?`

**Comment by birarda** [comment]
Cartfile

```
github "radex/SwiftyUserDefaults" "80b531d122b8ab2dca94e32dc8b497a7e495af99" # swifty wrapper for NSUserDefaults
github "Alamofire/Alamofire" ~> 3.0 # Elegant HTTP Networking
github "kishikawakatsumi/KeychainAccess" # wrapper for keychain access
github "tristanhimmelman/AlamofireObjectMapper" ~> 2.0 # Object mapping for REST API sanity
github "SwiftyJSON/SwiftyJSON" # easier JSON parsing for manual parsing
```

**Comment by acorscadden** [comment]
I've got the exact same issue @birarda. SDWebImage is failing iTunes connect validation, but it looks like bitcode is enabled. My frameworks also fail the otool bitcode test.

**Comment by abarnes** [comment]
I'm having this issue as well. I noticed my carthage/Build/watchOS frameworks have bitcode but tvOS doesn't (checking with "otool -l SwiftyJSON.framework/SwiftyJSON | grep LLVM" for example)

Cartfile: 
github "Alamofire/Alamofire"
github "SwiftyJSON/SwiftyJSON"

Carthage 0.10.0

**Comment by ikesyo** [comment]
> I noticed my carthage/Build/watchOS frameworks have bitcode but tvOS doesn't (checking with "otool -l SwiftyJSON.framework/SwiftyJSON | grep LLVM" for example)

`otool -l -arch arm64 SwiftyJSON.framework/SwiftyJSON | grep LLVM` would output expected result.

Are you using `copy-frameworks` for your watchOS/tvOS application target?

**Comment by ikesyo** [comment]
It might be related to this: #824

**Comment by abarnes** [comment]
I am using copy-frameworks as a run script in my build phases.

You were right, I can tell that bitcode is there when I use the command you provided:
otool -l -arch arm64 SwiftyJSON.framework/SwiftyJSON | grep LLVM

However, last time I tried to submit an app to iTunes connect using frameworks built by carthage, I was emailed an error from iTunes after uploading through the Application Loader:

Invalid Executable - The executable 'MyApp.app/Frameworks/SwiftyJSON.framework/SwiftyJSON' does not contain bitcode.

I was not able to resolve this until I manually build the frameworks and placed them into the Carthage/Build folder.

**Comment by NachoSoto** [comment]
@ikesyo it shouldn't be related to that. The only problem that needs solved in #824 is the `SUPPORTED_PLATFORMS` key.

**Comment by ScottPetit** [comment]
I think this can be closed now that #824 has been fixed and released in 0.11.  I was able to successfully submit with 0.11 after not being able to (due to the same error) with 0.10.
